### PR TITLE
Create script that checks markdown links

### PR DESCRIPTION
**Pull request by zachlatta**
Hopefully this'll be a good first step in noticing issues like #416 prematurely in the future.

### CODE REVIEW

**Comment by zachlatta** [comment]
Not ready for merge yet – I gotta play with the CircleCI config a bit.

**Comment by zachlatta** [comment]
Yes! There we go. That is a good failure.

File: circle.yml [hunk 1]
```diff
@@ -9,10 +6,21 @@ machine:
...
+    - ./meta/bin/check_markdown_links
+    - bundle exec rspec --color --require spec_helper spec --format progress:
+        pwd: unrejector
 
 deployment:
   hub:
     branch: master
     commands:
       - docker login -e $DOCKER_EMAIL -u $DOCKER_USER -p $DOCKER_PASS
-      - git clone git://github.com/hackedu/hack-camp /tmp/hack-camp && cd /tmp/hack-camp/unrejector && make push
```
Comment (maxwofford) [inline circle.yml:18]: What's the diff here?

File: cohort_4/playbook/workshops/portfolio/img/margin-challenge.md [hunk 2]
```diff
@@ -1,19 +0,0 @@
-#Challenge: Using margins to adjust elements
```
Comment (maxwofford) [inline cohort_4/playbook/workshops/portfolio/img/margin-challenge.md:1]: Why is this deleted?

File: circle.yml (hunk 1 above)
Comment (zachlatta) [inline circle.yml:18]: Added a newline at the end of the file. For some reason it wasn't there before.

I think it's a minor enough change that it's fine if it comes in this PR.

Comment (zachlatta) [inline circle.yml:18]: Also, just for reference, but newlines at the end of files is part of POSIX. See http://stackoverflow.com/questions/729692/why-should-files-end-with-a-newline

File: cohort_2/modules/week_2_student_outline.md [hunk 3]
```diff
@@ -192,4 +192,4 @@ See the documentation [here](http://meowth.servers.hackedu.us:8000/docs.md)
  - callAndSay
 
 **Write Clone**
-"Fork" the above JS Bin and work in there
\ No newline at end of file
+"Fork" the above JS Bin and work in there
```
Comment (maxwofford) [inline cohort_2/modules/week_2_student_outline.md:195]: What is the diff here?

File: cohort_4/playbook/workshops/portfolio/img/margin-challenge.md (hunk 2 above)
Comment (zachlatta) [inline cohort_4/playbook/workshops/portfolio/img/margin-challenge.md:1]: It's a duplicate of `cohort_4/playbook/workshops/portfolio/margin_challenge.md` and had broken links as a result of being in the `img` directory instead of `portfolio`.

File: cohort_2/modules/week_2_student_outline.md (hunk 3 above)
Comment (zachlatta) [inline cohort_2/modules/week_2_student_outline.md:195]: Same thing. New line at end of file.

File: cohort_4/playbook/workshops/portfolio/outlining_solution.md [hunk 4]
```diff
@@ -142,4 +142,4 @@ Part 2 is optional can be done at home at your liesure.
 
 ## Next
 
-[A bit more information about the structure of this tutorial](tutorial_format.md)
\ No newline at end of file
+[A bit more information about the structure of this tutorial](tutorial_format.md)
```
Comment (zachlatta) [inline cohort_4/playbook/workshops/portfolio/outlining_solution.md:145]: Also, for reference, same thing here. Newline at end of file.

File: cohort_4/playbook/workshops/portfolio/logos_solution.md [hunk 5]
```diff
@@ -140,4 +140,4 @@ Yay! Now it seems that all of the ements on the page are present!
 ## Next
 We've added EVERYTHING that we need in the HTML. Let's take your website and put it on the internet!
 
-[Put our code permanently on the live internet!](github.md)
\ No newline at end of file
+[Put our code permanently on the live internet!](github.md)
```
Comment (zachlatta) [inline cohort_4/playbook/workshops/portfolio/logos_solution.md:143]: Here too.

File: circle.yml (hunk 1 above)
Comment (maxwofford) [inline circle.yml:18]: Ah, thanks! Didn't see the red symbol on the line.

**Comment by maxwofford** [comment]
That's some crazy bash-fo you've got there @zachlatta. Really appreciate the script!
